pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -1055,8 +1057,11 @@ func TestHighPriorityFlushUnschedulableQLeftover(t *testing.T) {
 
 	addOrUpdateUnschedulablePod(q, &highPod)
 	addOrUpdateUnschedulablePod(q, &midPod)
+
+	q.lock.Lock()
 	q.unschedulableQ.podInfoMap[util.GetPodFullName(&highPod)].Timestamp = time.Now().Add(-1 * unschedulableQTimeInterval)
```

Comment:
`addOrUpdateUnschedulablePod` already does locking. Instead of applying locking here, I suggest to modify `addOrUpdateUnschedulablePod` to accept a PodInfo input instead of Pod, therefore you can first create a PodInfo and avoid updating the podInfoMap again (in L1062-1063).

 E.g.,
change
```go
func addOrUpdateUnschedulablePod(p *PriorityQueue, pod *v1.Pod) {
	p.lock.Lock()
	defer p.lock.Unlock()
	p.unschedulableQ.addOrUpdate(p.newPodInfo(pod))
}
```
to
```go
func addOrUpdateUnschedulablePod(p *PriorityQueue, podInfo *framework.PodInfo) {
	p.lock.Lock()
	defer p.lock.Unlock()
	p.unschedulableQ.addOrUpdate(podInfo)
}
```

Reply:
sgtm